tower-buffer/src/lib.rs

```diff
@@ -57,15 +59,51 @@ impl<E> ServiceError<E> {
     }
 }
 
-/// Errors produced by `Buffer`.
+/// The underlying `Service` is currently at capacity; wait for `poll_ready`.
 #[derive(Debug)]
-pub enum Error<E> {
-    /// The `Service` call errored.
-    Inner(E),
-    /// The underlying `Service` failed. All subsequent requests will fail.
-    Closed(Arc<ServiceError<E>>),
-    /// The underlying `Service` is currently at capacity; wait for `poll_ready`.
-    Full,
+pub struct FullError;
```

Comment:
TIOLI, but I would probably call this `Full` (and call `ClosedError` just `Closed`); IMO the impl of `Error` makes it pretty clear that this is an error...

Reply:
I am indifferent either way. 

That said, wouldn't the type be private after this change?

Reply:
@carllerche no, I don't think so, if the user wishes to downcast or use `error.source().is::<tower_buffer::Full>(), etc..

Reply:
I would probably put it in an `error` module... like https://docs.rs/tokio-sync/0.1.1/tokio_sync/mpsc/error/index.html

not sure if it should be `error` or `errors`.